OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
SLF4J(W): Class path contains multiple SLF4J providers.
SLF4J(W): Found provider [ch.qos.logback.classic.spi.LogbackServiceProvider@255eaa6b]
SLF4J(W): Found provider [org.apache.logging.slf4j.SLF4JServiceProvider@476fe690]
SLF4J(W): See https://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J(I): Actual provider is of type [ch.qos.logback.classic.spi.LogbackServiceProvider@255eaa6b]
15:23:41.590 [main] INFO fr.ca.cats.p1354.s3601.svcllm.utils.KafkaPropertiesLoader -- 
***** Missing application-kafka.properties for Consumer 


java.lang.NullPointerException
	at java.base/java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
	at java.base/java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
	at java.base/java.util.Properties.put(Properties.java:1301)
	at fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImpl.createConsumerFromProperties(ConsumerKafkaServiceImpl.java:128)
	at fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImpl.init(ConsumerKafkaServiceImpl.java:102)
	at fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImplTest.testInit(ConsumerKafkaServiceTest.java:48)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)


org.mockito.exceptions.misusing.NotAMockException: 
Argument passed to when() is not a mock!
Example of correct stubbing:
    doThrow(new RuntimeException()).when(mock).someMethod();

	at fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImplTest.testStartSuccess(ConsumerKafkaServiceTest.java:98)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)


org.mockito.exceptions.misusing.NotAMockException: 
Argument passed to when() is not a mock!
Example of correct stubbing:
    doThrow(new RuntimeException()).when(mock).someMethod();

	at fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImplTest.testStartFailure(ConsumerKafkaServiceTest.java:108)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

15:23:41.816 [main] ERROR fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImpl -- 
Exception raised from Kafka thread= 
Cannot enter synchronized block because "kafkaConsumer" is null
15:23:41.817 [main] INFO fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImpl -- 
Exception raised from Controller thread --> leaving Kafka thread 

15:23:41.817 [main] INFO fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImpl -- 
*** startKafka: kafkaConsumer is null; 

org.opentest4j.AssertionFailedError: Unexpected exception thrown: java.lang.NullPointerException: Cannot enter synchronized block because "kafkaConsumer" is null

	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:152)
	at org.junit.jupiter.api.AssertDoesNotThrow.createAssertionFailedError(AssertDoesNotThrow.java:84)
	at org.junit.jupiter.api.AssertDoesNotThrow.assertDoesNotThrow(AssertDoesNotThrow.java:53)
	at org.junit.jupiter.api.AssertDoesNotThrow.assertDoesNotThrow(AssertDoesNotThrow.java:36)
	at org.junit.jupiter.api.Assertions.assertDoesNotThrow(Assertions.java:3168)
	at fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImplTest.testStartKafkaHandlesException(ConsumerKafkaServiceTest.java:82)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.NullPointerException: Cannot enter synchronized block because "kafkaConsumer" is null
	at fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImpl.startKafka(ConsumerKafkaServiceImpl.java:182)
	at fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImplTest.lambda$testStartKafkaHandlesException$0(ConsumerKafkaServiceTest.java:82)
	at org.junit.jupiter.api.AssertDoesNotThrow.assertDoesNotThrow(AssertDoesNotThrow.java:49)
	... 6 more

15:23:42.225 [main] INFO fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImpl -- 
*** startKafkaCore() :  isActive= true


java.lang.NullPointerException: Cannot invoke "fr.ca.cats.p1354.DemandeSuggestionValue.getMailId()" because "this.suggestionValue" is null

	at fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImpl.startKafkaCore(ConsumerKafkaServiceImpl.java:211)
	at fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImplTest.testStartKafkaCoreHandlesLLMError(ConsumerKafkaServiceTest.java:90)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)


org.mockito.exceptions.misusing.NotAMockException: 
Argument passed to when() is not a mock!
Example of correct stubbing:
    doThrow(new RuntimeException()).when(mock).someMethod();

	at fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImplTest.testScheduleStartHandlesException(ConsumerKafkaServiceTest.java:118)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

15:23:42.311 [main] ERROR fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImpl -- 
Exception raised from Kafka thread= 
Cannot enter synchronized block because "kafkaConsumer" is null
15:23:42.312 [main] INFO fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImpl -- 
Exception raised from Controller thread --> leaving Kafka thread 

15:23:42.312 [main] INFO fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImpl -- 
*** startKafka: kafkaConsumer is null; 

java.lang.NullPointerException: Cannot enter synchronized block because "kafkaConsumer" is null

	at fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImpl.startKafka(ConsumerKafkaServiceImpl.java:182)
	at fr.ca.cats.p1354.s3601.svcllm.service.impl.ConsumerKafkaServiceImplTest.testStartKafkaWhenActive(ConsumerKafkaServiceTest.java:73)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)


Process finished with exit code -1
